config/config.go

```diff
@@ -0,0 +1,73 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+type Config struct {
+	HomeDir string
+	BinDir  string
+	VarDir  string
+	RunDir  string
+
+	Telemetry struct {
+		Disable bool
+	}
+}
+
+type contextKey string
+
+var configContextKey = contextKey("config")
+
+func LoadDefault(cfg *Config) error {
+	if cfg.HomeDir == "" {
+		cfg.HomeDir = exoHome()
+	}
+
+	// TODO: make directory if necessary.
+	configFile := getExoPath(cfg.HomeDir, "config.toml", "EXO_CONFIG")
+	if err := loadFromFile(configFile, cfg); err != nil {
+		return err
+	}
+
+	if cfg.BinDir == "" {
+		cfg.BinDir = getExoPath(cfg.HomeDir, "bin", "EXO_BIN")
+	}
+	if cfg.RunDir == "" {
+		cfg.RunDir = getExoPath(cfg.HomeDir, "run", "EXO_RUN")
+	}
+	if cfg.VarDir == "" {
+		cfg.VarDir = getExoPath(cfg.HomeDir, "var", "EXO_VAR")
+	}
+
+	return nil
+}
+
+func MustLoadDefault(cfg *Config) {
+	if err := LoadDefault(cfg); err != nil {
+		panic(err)
+	}
+}
+
+func loadFromFile(filePath string, cfg *Config) error {
+	_, err := toml.DecodeFile(filePath, cfg)
+	if os.IsNotExist(err) {
+		if err = os.WriteFile(filePath, []byte{'\n'}, 0644); err != nil {
+			return fmt.Errorf("creating initial config file: %w", err)
```

Comment:
Is this actually an error condition or is it success where cfg is expected to be set to a zero value?

Reply:
I think this should be an error condition. The zero value *should* be useful, but if we try to use a config file that we can't access, I think that should be an error rather than a silent failure.